### ISSUE TITLE
[JENKINS-30231] Build creates second workspace@2 for non-concurrent build configuration

### ISSUE THREAD

**Issue by jenkins-infra-bot** [timja-org/jenkins-gh-issues-poc-11-15#10037]
<p>I didn't have this issue when I was using ver 1.580.3.<br/>
After upgrading Jenkins ver 1.609.1 (together with other Jenkins plugins update),<br/>
I started to see this issue.</p>

<p>My jenkins job is using multi-config project type (aka Matrix project),<br/>
and has 3 variants.<br/>
I didn't set to be run concurrently,<br/>
so the 3 variants will start the build one by one.</p>

<p>In the previous version 1.580.3,<br/>
it will reuse the same workspace folder for all of 3 variant builds.</p>

<p>In the new version 1.609.1,<br/>
when multiple jobs are triggerd,<br/>
I can see sometimes the first variant build of the second job<br/>
will start to create and use workspace@​2<br/>
instead of re-using original worksapce.</p>

<p>Note:<br/>
I'm also using disk-usage plugin.<br/>
At the end of build,<br/>
it will trigger that plugin to calculate workspace disk space.<br/>
My work space tends to be large and sometimes it take several minutes to finish.</p>

<p>I suspect it's that disk-usage plugin still occupying the original worksapce,<br/>
but the second job in queue starts earlier,<br/>
so it creates workspace@​2 for the first variable build.</p>

<p>From workspace@​2,<br/>
I always see only one variant only under AXIS folder.</p>

---
<details><summary><i>Originally reported by <a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=totoroliu">totoroliu</a>, imported from: <a href="https://issues.jenkins.io/browse/JENKINS-30231" target="_blank">Build creates second workspace@2 for non-concurrent build configuration</a></i></summary>
<i><ul>
<li><b>assignee</b>: <img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/dnusbaum.png" title="dnusbaum's avatar" /> <a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=dnusbaum">dnusbaum</a>
<li><b>status</b>: Resolved
<li><b>priority</b>: Minor
<li><b>component(s)</b>: core, matrix-project-plugin
<li><b>resolution</b>: Duplicate
<li><b>resolved</b>: 2018-09-24T13:03:46+00:00
<li><b>votes</b>: 4
<li><b>watchers</b>: 22
<li><b>imported</b>: 2025-11-16
</ul></i>
<details><summary>Raw content of original issue</summary>

<pre>
<p>I didn't have this issue when I was using ver 1.580.3.
After upgrading Jenkins ver 1.609.1 (together with other Jenkins plugins update),
I started to see this issue.</p>

<p>My jenkins job is using multi-config project type (aka Matrix project),
and has 3 variants.
I didn't set to be run concurrently,
so the 3 variants will start the build one by one.</p>

<p>In the previous version 1.580.3,
it will reuse the same workspace folder for all of 3 variant builds.</p>

<p>In the new version 1.609.1,
when multiple jobs are triggerd,
I can see sometimes the first variant build of the second job
will start to create and use workspace@2
instead of re-using original worksapce.</p>

<p>Note:
I'm also using disk-usage plugin.
At the end of build,
it will trigger that plugin to calculate workspace disk space.
My work space tends to be large and sometimes it take several minutes to finish.</p>

<p>I suspect it's that disk-usage plugin still occupying the original worksapce,
but the second job in queue starts earlier,
so it creates workspace@2 for the first variable build.</p>

<p>From workspace@2,
I always see only one variant only under AXIS folder.</p></pre>
</details>
</details>
<details><summary><i>environment</i></summary>

```
OS: Ubuntu 12.04 64-bit<br/>
Jenkins: 1.609.1 (upgraded from 1.580.3)<br/>
disk-usage plugin: 0.25 (upgraded from 0.24)<br/>
Matrix Project Plugin: 1.6 (upgraded from 1.4.1)
```
</details>
<details><summary><i>3 attachments</i></summary>

- [image-2020-08-16-13-41-37-491.png](https://issues.jenkins.io/secure/attachment/52222/image-2020-08-16-13-41-37-491.png)
  > ![image-2020-08-16-13-41-37-491.png](https://issues.jenkins.io/secure/attachment/52222/image-2020-08-16-13-41-37-491.png)
- [image-2020-08-16-13-42-26-759.png](https://issues.jenkins.io/secure/attachment/52223/image-2020-08-16-13-42-26-759.png)
  > ![image-2020-08-16-13-42-26-759.png](https://issues.jenkins.io/secure/attachment/52223/image-2020-08-16-13-42-26-759.png)
- [image-2020-08-16-13-43-23-877.png](https://issues.jenkins.io/secure/attachment/52224/image-2020-08-16-13-43-23-877.png)
  > ![image-2020-08-16-13-43-23-877.png](https://issues.jenkins.io/secure/attachment/52224/image-2020-08-16-13-43-23-877.png)
</details>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i>totoroliu's <a href="https://issues.jenkins.io/browse/JENKINS-30231?focusedId=235779&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-235779">comment</a>:</i></summary>


<details><summary><sub><i>Raw content of original comment:</i></sub></summary>

<pre>
<p>Per comment from Jesse Glick,
I create this separate issue
which has similar symptom.</p>
</pre>
</details>

</details>
<p>Per comment from Jesse Glick,<br/>
I create this separate issue<br/>
which has similar symptom.</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/will177.svg" title="will177's avatar" /> will177's <a href="https://issues.jenkins.io/browse/JENKINS-30231?focusedId=273076&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-273076">comment</a>:</i></summary>


<details><summary><sub><i>Raw content of original comment:</i></sub></summary>

<pre>
<p>I am seeing this issue today on one of my pipelines.  Is there a quick fix workaround I can use to get rid of it and reset the pipeline?  I don't mind starting the pipeline from scratch again - just wondering how to get rid of the @2 workspace.  Anyone know?</p>
</pre>
</details>

</details>
<p>I am seeing this issue today on one of my pipelines.  Is there a quick fix workaround I can use to get rid of it and reset the pipeline?  I don't mind starting the pipeline from scratch again - just wondering how to get rid of the @​2 workspace.  Anyone know?</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/mksmith.png" title="mksmith's avatar" /> mksmith's <a href="https://issues.jenkins.io/browse/JENKINS-30231?focusedId=277463&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-277463">comment</a>:</i></summary>


<details><summary><sub><i>Raw content of original comment:</i></sub></summary>

<pre>
<p><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=will177" class="user-hover" rel="will177">will177</a> It looks like this might be a solution: <a href="https://issues.jenkins-ci.org/browse/JENKINS-14354" class="external-link" target="_blank" rel="nofollow noopener">https://issues.jenkins-ci.org/browse/JENKINS-14354</a></p>

<p>It looks like you can replace the "@" separator with a separator of your choice. </p>
</pre>
</details>

</details>
<p><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=will177" class="user-hover" rel="will177">will177</a> It looks like this might be a solution: <a href="https://issues.jenkins-ci.org/browse/JENKINS-14354" class="external-link" target="_blank" rel="nofollow noopener">https://issues.jenkins-ci.org/browse/JENKINS-14354</a></p>

<p>It looks like you can replace the "@" separator with a separator of your choice. </p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i>totoroliu's <a href="https://issues.jenkins.io/browse/JENKINS-30231?focusedId=277465&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-277465">comment</a>:</i></summary>


<details><summary><sub><i>Raw content of original comment:</i></sub></summary>

<pre>
<p>I use customize worksapce configuration to hardcoded the workspace name to work-around this problem</p>
</pre>
</details>

</details>
<p>I use customize worksapce configuration to hardcoded the workspace name to work-around this problem</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i>rg's <a href="https://issues.jenkins.io/browse/JENKINS-30231?focusedId=316699&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-316699">comment</a>:</i></summary>


<details><summary><sub><i>Raw content of original comment:</i></sub></summary>

<pre>
<p>This is still a problem. We restrict our jobs to one per node and still see a number of @2 workspaces. Replacing @ with _ (like <a href="https://issues.jenkins.io/browse/JENKINS-14354" title="Concurrent build jobs use BAD naming convention for workspaces" class="issue-link" data-issue-key="JENKINS-14354"><del>JENKINS-14354</del></a>) helps builds which get tripped up by @ but we still have duplication. As we have a number of large workspaces this wastes a lot of diskspace.</p>
</pre>
</details>

</details>
<p>This is still a problem. We restrict our jobs to one per node and still see a number of @​2 workspaces. Replacing @ with _ (like <a href="https://issues.jenkins.io/browse/JENKINS-14354" title="Concurrent build jobs use BAD naming convention for workspaces" class="issue-link" data-issue-key="JENKINS-14354"><del>JENKINS-14354</del></a>) helps builds which get tripped up by @ but we still have duplication. As we have a number of large workspaces this wastes a lot of diskspace.</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/j1mehta.png" title="j1mehta's avatar" /> j1mehta's <a href="https://issues.jenkins.io/browse/JENKINS-30231?focusedId=370740&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-370740">comment</a>:</i></summary>


<details><summary><sub><i>Raw content of original comment:</i></sub></summary>

<pre>
<p><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=dnusbaum" class="user-hover" rel="dnusbaum">dnusbaum</a>, thank you for the reply. Makes sense, will do that. However, it will take some time since this problem occurs in a Jenkins environment of which I am not an admin and would need to replicate it locally. Meanwhile, if I could get any pointers, that would be great. </p>

<p>EDIT: It doesn't get reproduced right away. It starts occurring only after several builds and hence is kinda hard to reproduce.</p>
</pre>
</details>

</details>
<p><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=dnusbaum" class="user-hover" rel="dnusbaum">dnusbaum</a>, thank you for the reply. Makes sense, will do that. However, it will take some time since this problem occurs in a Jenkins environment of which I am not an admin and would need to replicate it locally. Meanwhile, if I could get any pointers, that would be great. </p>

<p>EDIT: It doesn't get reproduced right away. It starts occurring only after several builds and hence is kinda hard to reproduce.</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i>kamil02's <a href="https://issues.jenkins.io/browse/JENKINS-30231?focusedId=370756&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-370756">comment</a>:</i></summary>


<details><summary><sub><i>Raw content of original comment:</i></sub></summary>

<pre>
<p><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=j1mehta" class="user-hover" rel="j1mehta">j1mehta</a> <a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=dnusbaum" class="user-hover" rel="dnusbaum">dnusbaum</a> I think we triggered this issue by removing the working directory of the job within the workspace of a node by hand. </p>

<p>Something like </p>
<ol>
	<li>Run job_a on node b</li>
	<li>let it finish</li>
	<li>login via ssh onto node b</li>
	<li>rm -R {JENKINS_WORKSPACE}/job_a</li>
	<li>rerun job a</li>
	<li>new folder with @2 suffix should be created</li>
</ol>


<p>Unfortunately i cant evaluate this right now.</p>
</pre>
</details>

</details>
<p><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=j1mehta" class="user-hover" rel="j1mehta">j1mehta</a> <a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=dnusbaum" class="user-hover" rel="dnusbaum">dnusbaum</a> I think we triggered this issue by removing the working directory of the job within the workspace of a node by hand. </p>

<p>Something like </p>
<ol>
	<li>Run job_a on node b</li>
	<li>let it finish</li>
	<li>login via ssh onto node b</li>
	<li>rm -R {JENKINS_WORKSPACE}/job_a</li>
	<li>rerun job a</li>
	<li>new folder with @​2 suffix should be created</li>
</ol>


<p>Unfortunately i cant evaluate this right now.</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i>kamil02's <a href="https://issues.jenkins.io/browse/JENKINS-30231?focusedId=370822&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-370822">comment</a>:</i></summary>


<details><summary><sub><i>Raw content of original comment:</i></sub></summary>

<pre>
<p><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=j1mehta" class="user-hover" rel="j1mehta">j1mehta</a> did you also try to restart the jenkins master after the deletion of both (original and @2) workspaces? If i remember correctly, i also had the feeling, that jenkins uses some kind of cache to remember where the workspace is supposed to be. As mentioned before, restarting the jenkins master instance after deleting the original and @2 workspace solved our problem.</p>
</pre>
</details>

</details>
<p><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=j1mehta" class="user-hover" rel="j1mehta">j1mehta</a> did you also try to restart the jenkins master after the deletion of both (original and @​2) workspaces? If i remember correctly, i also had the feeling, that jenkins uses some kind of cache to remember where the workspace is supposed to be. As mentioned before, restarting the jenkins master instance after deleting the original and @​2 workspace solved our problem.</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/adamkolarik.png" title="adamkolarik's avatar" /> adamkolarik's <a href="https://issues.jenkins.io/browse/JENKINS-30231?focusedId=377290&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-377290">comment</a>:</i></summary>


<details><summary><sub><i>Raw content of original comment:</i></sub></summary>

<pre>
<p>I'm seeing this issue on version 2.176.1. We also don't have concurrent or Matrix builds setup, just one executor per node. On one specific node, we got the workspace@2 duplicated workspace. I tried deleting both workspaces on the node manually and restart Jenkins server to see if that helps and it did.</p>
</pre>
</details>

</details>
<p>I'm seeing this issue on version 2.176.1. We also don't have concurrent or Matrix builds setup, just one executor per node. On one specific node, we got the workspace@​2 duplicated workspace. I tried deleting both workspaces on the node manually and restart Jenkins server to see if that helps and it did.</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i>kamil02's <a href="https://issues.jenkins.io/browse/JENKINS-30231?focusedId=384880&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-384880">comment</a>:</i></summary>


<details><summary><sub><i>Raw content of original comment:</i></sub></summary>

<pre>
<p>Today the same error occurred again: Jenkins created a ..@2 and ...@3 folder in the .../workspace directory. I had to delete these via SSH and restart the master =&gt; back to normal job workspace again.  </p>

<p>Jenkins ver. 2.176.3</p>
</pre>
</details>

</details>
<p>Today the same error occurred again: Jenkins created a ..@​2 and ...@​3 folder in the .../workspace directory. I had to delete these via SSH and restart the master => back to normal job workspace again.  </p>

<p>Jenkins ver. 2.176.3</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/xianpeng.png" title="xianpeng's avatar" /> xianpeng's <a href="https://issues.jenkins.io/browse/JENKINS-30231?focusedId=388034&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-388034">comment</a>:</i></summary>


<details><summary><sub><i>Raw content of original comment:</i></sub></summary>

<pre>
<p>I have the same issue on Jenkins ver. 2.176.3.</p>
</pre>
</details>

</details>
<p>I have the same issue on Jenkins ver. 2.176.3.</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i>naveen6s's <a href="https://issues.jenkins.io/browse/JENKINS-30231?focusedId=391659&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-391659">comment</a>:</i></summary>


<details><summary><sub><i>Raw content of original comment:</i></sub></summary>

<pre>
<p>I have the same issue with Jenkins version 2.222.3</p>

<p>I have admin access, so I rebooted Jenkins master after deleting ..@2 folder. This did not help either.</p>
</pre>
</details>

</details>
<p>I have the same issue with Jenkins version 2.222.3</p>

<p>I have admin access, so I rebooted Jenkins master after deleting ..@​2 folder. This did not help either.</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i>olssons's <a href="https://issues.jenkins.io/browse/JENKINS-30231?focusedId=395803&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-395803">comment</a>:</i></summary>


<details><summary><sub><i>Raw content of original comment:</i></sub></summary>

<pre>
<p>I also have this problem.  I checkout two repositories into my workspace and often times Jenkins creates a second workspace for the parallel-running job which checks out the second repo.  The tools assume they are in the same workspace and promptly fail.  Possible workaround might be to save the location of WORKSPACE in the beginning and then with each parallel task, always change directory (cd) back to that location before doing anything.  </p>

<p>For example,</p>

<p><span class="image-wrap" style=""><img src="https://issues.jenkins.io/secure/attachment/52223/52223_image-2020-08-16-13-42-26-759.png" height="154" width="509" style="border: 0px solid black" /></span></p>

<p><span class="image-wrap" style=""><img src="https://issues.jenkins.io/secure/attachment/52224/52224_image-2020-08-16-13-43-23-877.png" height="209" width="728" style="border: 0px solid black" /></span></p>
</pre>
</details>

</details>
<p>I also have this problem.  I checkout two repositories into my workspace and often times Jenkins creates a second workspace for the parallel-running job which checks out the second repo.  The tools assume they are in the same workspace and promptly fail.  Possible workaround might be to save the location of WORKSPACE in the beginning and then with each parallel task, always change directory (cd) back to that location before doing anything.  </p>

<p>For example,</p>

<p><span class="image-wrap" style=""><img src="https://issues.jenkins.io/secure/attachment/52223/52223_image-2020-08-16-13-42-26-759.png" height="154" width="509" style="border: 0px solid black" /></span></p>

<p><span class="image-wrap" style=""><img src="https://issues.jenkins.io/secure/attachment/52224/52224_image-2020-08-16-13-43-23-877.png" height="209" width="728" style="border: 0px solid black" /></span></p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i>olssons's <a href="https://issues.jenkins.io/browse/JENKINS-30231?focusedId=395804&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-395804">comment</a>:</i></summary>


<details><summary><sub><i>Raw content of original comment:</i></sub></summary>

<pre>
<p><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=dnusbaum" class="user-hover" rel="dnusbaum">dnusbaum</a> - I see that the issue is marked as Resolved.  In this case, should a new issue be created or the preference is to Reopen?  I didn't want to reopen if it really needs to be a new issue instead.</p>
</pre>
</details>

</details>
<p><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=dnusbaum" class="user-hover" rel="dnusbaum">dnusbaum</a> - I see that the issue is marked as Resolved.  In this case, should a new issue be created or the preference is to Reopen?  I didn't want to reopen if it really needs to be a new issue instead.</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/dnusbaum.png" title="dnusbaum's avatar" /> dnusbaum's <a href="https://issues.jenkins.io/browse/JENKINS-30231?focusedId=395845&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-395845">comment</a>:</i></summary>


<details><summary><sub><i>Raw content of original comment:</i></sub></summary>

<pre>
<p><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=olssons" class="user-hover" rel="olssons">olssons</a> My guess is that there are probably many distinct issues being reported here (e.g. some scenarios are likely specific to (multibranch) pipelines, but there were also bugs fixed that affected freestyle and other project types). I would go ahead and open a new issue with full details on your specific case, ideally with steps on how to reproduce the problem from scratch.</p>
</pre>
</details>

</details>
<p><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=olssons" class="user-hover" rel="olssons">olssons</a> My guess is that there are probably many distinct issues being reported here (e.g. some scenarios are likely specific to (multibranch) pipelines, but there were also bugs fixed that affected freestyle and other project types). I would go ahead and open a new issue with full details on your specific case, ideally with steps on how to reproduce the problem from scratch.</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i>olssons's <a href="https://issues.jenkins.io/browse/JENKINS-30231?focusedId=396202&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-396202">comment</a>:</i></summary>


<details><summary><sub><i>Raw content of original comment:</i></sub></summary>

<pre>
<p><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=dnusbaum" class="user-hover" rel="dnusbaum">dnusbaum</a> - With my issue, the double workspace never occurs during the "Build Software" parallel tasks.  However, when I have 3 parallel tasks in a row, like under "Analysis and Adaptation", this is where the problem <b>sometimes</b> happens.  I say <b>sometimes</b> because it happens maybe half the time or 3 out of 4 builds in a row. </p>

<p>I first started my workarounds by removing the second checkout like I mentioned before (see "Checkout tools" above).  That did not fix the problem.  I then made all the tasks in Analysis and Adaptation serial.  That (obviously) fixed the problem.  If I were to guess, it didn't like 3 tasks running in parallel, all using the same workspace.  My builds take about 10-15 minutes longer now, but no more failures at least.</p>

<p>I may open an issue in the future. </p>

<p>PS - I could never recreate the problem from the Jenkins Pipeline script, only from a checked in Jenkinsfile itself</p>
</pre>
</details>

</details>
<p><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=dnusbaum" class="user-hover" rel="dnusbaum">dnusbaum</a> - With my issue, the double workspace never occurs during the "Build Software" parallel tasks.  However, when I have 3 parallel tasks in a row, like under "Analysis and Adaptation", this is where the problem <b>sometimes</b> happens.  I say <b>sometimes</b> because it happens maybe half the time or 3 out of 4 builds in a row. </p>

<p>I first started my workarounds by removing the second checkout like I mentioned before (see "Checkout tools" above).  That did not fix the problem.  I then made all the tasks in Analysis and Adaptation serial.  That (obviously) fixed the problem.  If I were to guess, it didn't like 3 tasks running in parallel, all using the same workspace.  My builds take about 10-15 minutes longer now, but no more failures at least.</p>

<p>I may open an issue in the future. </p>

<p>PS - I could never recreate the problem from the Jenkins Pipeline script, only from a checked in Jenkinsfile itself</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i>wizzzozzz's <a href="https://issues.jenkins.io/browse/JENKINS-30231?focusedId=426526&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-426526">comment</a>:</i></summary>


<details><summary><sub><i>Raw content of original comment:</i></sub></summary>

<pre>
<p>Is there anyway to disable this option?</p>

<p>so no new workspace will be created and only the original workspace will be used alone? (I'm aware of the risks)</p>
</pre>
</details>

</details>
<p>Is there anyway to disable this option?</p>

<p>so no new workspace will be created and only the original workspace will be used alone? (I'm aware of the risks)</p>

**Comment by jenkins-infra-bot** [comment]
<i>[Originally related to: <a href="https://github.com/timja-org/jenkins-gh-issues-poc-11-15/issues?q=in%3Atitle%20JENKINS-21622">JENKINS-21622</a>]</i>
